lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -781,24 +781,48 @@ SILCombiner::visitInjectEnumAddrInst(InjectEnumAddrInst *IEAI) {
 
   // Ok, we have a payload enum, make sure that we have a store previous to
   // us...
-  SILBasicBlock::iterator II = IEAI->getIterator();
+  auto allocStackOperand = IEAI->getOperand();
```

Comment:
You need to check if the operand is really an AllocStackInst and if the found store,init_enum_data_addr,inject_enum_addr are the only instructions which write to this location or may escape the stack address.
This is required as you are not doing a dataflow analysis and you cannot be sure if the IEAI address may be overwritten between the init_enum_data_addr and inject_enum_addr.